x/datadeal/types/deal.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+func NewDeal(dealID uint64, deal Deal) Deal {
+
+	dealAddress := NewDealAddress(dealID)
+
+	return Deal{
+		Id:           dealID,
+		Address:      dealAddress.String(),
+		DataSchema:   deal.GetDataSchema(),
+		Budget:       deal.GetBudget(),
+		MaxNumData:   deal.GetMaxNumData(),
+		CurNumData:   0,
+		BuyerAddress: deal.GetBuyerAddress(),
+		Status:       DEAL_STATUS_ACTIVE,
+	}
+}
+
+func NewDealAddress(dealID uint64) sdk.AccAddress {
+	dealKey := "deal" + strconv.FormatUint(dealID, 10)
+	return authtypes.NewModuleAddress(dealKey)
+}
+
+func (m Deal) ValidateBasic() error {
```

Comment:
Let's validate `CurNumData` as well :)

Reply:
Thanks, I add it

Reply:
And also, it would be better to check the `CurNumData` is greater than 0 like `MaxNumData`

Reply:
Thanks, I forgot. I will add it to next PR.